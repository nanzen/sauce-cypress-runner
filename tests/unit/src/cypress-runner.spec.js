jest.mock('cypress');
jest.mock('../../../src/sauce-reporter');

const cypress = require('cypress');
const path = require('path');
const { cypressRunner } = require('../../../src/cypress-runner');
const { sauceReporter, prepareAssets } = require('../../../src/sauce-reporter');

describe('.cypressRunner', function () {
  let mockCypressRun, mockSauceReporter, originalEnv;
  beforeEach(function () {
    originalEnv = process.env;
    process.env.SAUCE_BUILD_NAME = 'fake-build';
    mockCypressRun = cypress.run.mockImplementation(() => ({
      failures: 0,
    }));
    prepareAssets.mockImplementation(() => (
      []
    ));
    mockSauceReporter = sauceReporter.mockImplementation(() => {});
  });
  afterEach(function () {
    process.env = originalEnv;
  });
  it('fixtures/kitchen-sink.json', async function () {
    const statusOne = await cypressRunner(path.join(__dirname, 'fixtures', 'kitchen-sink.json'), 'foo');
    const statusTwo = await cypressRunner(path.join(__dirname, 'fixtures', 'kitchen-sink.json'), 'bar');
    expect(statusOne).toEqual(true);
    expect(statusTwo).toEqual(true);
    expect(mockCypressRun.mock.calls).toMatchSnapshot();
    expect(mockSauceReporter.mock.calls).toMatchSnapshot();
  });
});