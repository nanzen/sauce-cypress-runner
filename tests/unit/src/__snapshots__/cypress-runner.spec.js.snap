// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.cypressRunner fixtures/kitchen-sink.json 1`] = `
Array [
  Array [
    Object {
      "browser": "chrome",
      "config": Object {
        "reporter": "/Users/danielgraham/sauce-cypress-runner/src/custom-reporter.js",
        "reporterOptions": Object {
          "mochaFile": "__assets__/[suite].xml",
          "specFolder": "__assets__/",
          "specRoot": "cypress/integration",
        },
        "screenshotsFolder": "__assets__",
        "video": true,
        "videoCompression": false,
        "videoUploadOnPasses": false,
        "videosFolder": "__assets__",
        "viewportHeight": 100,
        "viewportWidth": 200,
      },
      "configFile": "alternative-cypress.json",
      "name": "foo",
      "project": "/path/to/cypress/project",
      "resultsFolder": "__assets__",
    },
  ],
  Array [
    Object {
      "browser": "firefox",
      "config": Object {
        "reporter": "/Users/danielgraham/sauce-cypress-runner/src/custom-reporter.js",
        "reporterOptions": Object {
          "mochaFile": "__assets__/[suite].xml",
          "specFolder": "__assets__/",
          "specRoot": "cypress/integration",
        },
        "screenshotsFolder": "__assets__",
        "video": true,
        "videoCompression": false,
        "videoUploadOnPasses": false,
        "videosFolder": "__assets__",
        "viewportHeight": 300,
        "viewportWidth": 400,
      },
      "configFile": "alternative-cypress.json",
      "name": "bar",
      "project": "/path/to/cypress/project",
      "resultsFolder": "__assets__",
    },
  ],
]
`;

exports[`.cypressRunner fixtures/kitchen-sink.json 2`] = `
Array [
  Array [
    "fake-build",
    "chrome",
    Array [],
    0,
  ],
  Array [
    "fake-build",
    "firefox",
    Array [],
    0,
  ],
]
`;
